mithril-common/src/entities/signed_entity_type.rs

```diff
@@ -101,5 +101,15 @@ impl SignedEntityType {
     }
 }
 
+impl SignedEntityTypeDiscriminants {
+    /// Get the database value from enum's instance
+    pub fn index(&self) -> usize {
```

Comment:
This method should be used to compute the `pub fn index(&self) -> usize` for the `SignedEntityType` to avoid match duplication IMO

Reply:
Not sure how to do it properly without prior discussions. Maybe this should be in a separate PR.